hostap: add WPS PBC and PIN for AP mode

MaochenWang1 opened the pull request:
add WPS PBC and PIN for AP mode.

MaochenWang1 committed:
hostap: add WPS PBC and PIN for AP mode

add WPS PBC and PIN for AP mode.

Signed-off-by: Maochen Wang <maochen.wang@nxp.com>

MaochenWang1 commented:
Hi @jukkar @krish2718 please review

krish2718 commented on modules/hostap/CMakeLists.txt:
```diff
@@ -215,8 +215,8 @@ zephyr_library_compile_definitions_ifdef(CONFIG_WIFI_NM_WPA_SUPPLICANT_AP
 	CONFIG_IEEE80211AX
-	CONFIG_EAP_SERVER
-	CONFIG_EAP_SERVER_IDENTITY
+	EAP_SERVER
+	EAP_SERVER_IDENTITY
```
better to use a separate commit to fix this.

MaochenWang1 replied:
yes, will separate

krish2718 commented on include/zephyr/net/wifi_mgmt.h:
```diff
@@ -1303,6 +1310,14 @@ struct wifi_mgmt_ops {
+	 *
+	 * @return 0 if ok, < 0 if error
+	 */
+	int (*ap_wps_config)(const struct device *dev, struct wifi_wps_config_params *params);
```
the structure `wifi_wps_config_params` is same, so, why not use existing `wps_config` for AP too?

MaochenWang1 replied:
yes, can use the same

krish2718 commented on modules/hostap/src/supp_api.c:
```diff
@@ -1550,6 +1550,106 @@ int supplicant_ap_config_params(const struct device *dev, struct wifi_ap_config_
 }
 #endif
 
+#ifdef CONFIG_WIFI_NM_HOSTAPD_WPS
```
Even these functions are identical to WPA supp based WPS except for using hostapd_cli, so, makes sense to avoid duplication.

MaochenWang1 replied:
not only the hostapd_cli, also get_hostapd_handle is different. Separate the WPS case for STA and SAP is more clean way,  as in the future, the hostapd code may be split into another file.

MaochenWang1 replied:
Hi @krish2718  there are still coming PRs from our downstream, better add supp_api_wps.c... in the future. thanks

krish2718 replied:
Sure, I was only mentioning it for the future, we already have a task https://github.com/zephyrproject-rtos/zephyr/issues/77383 we can keep adding comments there so that we don't miss.

krish2718 reviewed (commented):
also maybe the file `supp_api.c` is becoming large, so, may be it's time to start adding `supp_api_wps.c`, `supp_api_p2p.c` etc?

MaochenWang1 committed:
hostap: use the correct config for EAP_SERVER

Should use EAP_SERVER and EAP_SERVER_IDENTITY as the correct config,
for the WPS case of AP.

Signed-off-by: Maochen Wang <maochen.wang@nxp.com>

MaochenWang1 commented:
Hi @jukkar  please review

MaochenWang1 commented:
Hi @krish2718 @jukkar @dleach02 please review

henrikbrixandersen commented:
Please rebase on `main` and resolve any merge conflicts.

dleach02 reviewed (changes requested):
temporary hold until #79536 merges

MaochenWang1 commented:
Hi @krish2718 @jukkar @dleach02 please review

MaochenWang1 commented:
Hi @krish2718 @jukkar @dleach02 please review

MaochenWang1 commented:
Hi @krish2718 @dleach02 fix the conflict, please review again

MaochenWang1 commented:
Hi @krish2718 @jukkar @dleach02 fix the conflict, please review again

MaochenWang1 commented:
Hi @dleach02  please review and approve this

MaochenWang1 commented:
Hi @henrikbrixandersen rebased and conflicts resolved, can you help review again?

henrikbrixandersen commented:
> temporary hold until #79536 merges

#79536 was merged. Please take another look.

MaochenWang1 commented:
Hi @henrikbrixandersen 
David was worried that this PR may cause conflict on https://github.com/zephyrproject-rtos/zephyr/pull/79536, which is also from me and more urgent, so he commented on this.
Now since https://github.com/zephyrproject-rtos/zephyr/pull/79536 was merged, and I rebased and fixed the conflict, I think can go ahead and merge this PR. Thanks.

dleach02 reviewed (dismissed):
blocking PR merged
